docs/guides/implementation-guide.mdx

```diff
@@ -24,8 +24,22 @@ If you would like to dive deeper, check out some of the Hyperlane implementation
 
 Below describes the onchain contract spec for the Hyperlane protocol. It uses solidity types for familiarity but everything should be generalizable to other languages.
 
-- `address` should be interpreted as the local chain's address type
-- `payable` describes a function that allows callers to pass native tokens
+- `address` should be interpreted as the local chain's address type.
+- `payable` describes a function that allows callers to pass native tokens.
+- Ensure events are properly emitted for all important state changes.
+- Clearly document any deviations from the reference implementation and their rationale.
+
+### Some things to consider
+
+**Byte representation:**
+
+- Be aware that different chains may have different native types for representing bytes. For example, StarkNet uses felt252 for contract addresses, which may not fully accommodate 32-byte addresses used in other chains.
```

Comment:
> which may not fully accommodate 32-byte addresses used in other chains.

Im not exactly sure what is meant by this but iiuc `sizeof(felt252) <= sizeof(bytes32)` so this feels slighlty misleading
ethereum addresses are also 20 bytes <= 32 bytes

Reply:
sizeof(felt252) = 252 bits and  sizeof(bytes32) = 8 x 32 = 256 bits, but yes for EVM it doesn't matter because addresses are 160 bits.